service-registry/src/main/java/org/apache/servicecomb/serviceregistry/consumer/MicroserviceVersions.java

```diff
@@ -118,7 +118,25 @@ public void pullInstances() {
     List<MicroserviceInstance> pulledInstances = microserviceInstances.getInstancesResponse().getInstances();
     String rev = microserviceInstances.getRevision();
 
-    setInstances(pulledInstances, rev);
+    safeSetInstances(pulledInstances, rev);
+  }
+
+  protected void safeSetInstances(List<MicroserviceInstance> pulledInstances, String rev) {
+    try {
+      setInstances(pulledInstances, rev);
```

Comment:
In which situation will this fail?

Reply:
like dynamic create class failed

Reply:
or failed to request from sc 
if happened, exception eat by scheduled thread pool, we can not know there is a exception.